Api framework graphql interfaces mark

mrtuvn opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Mark GraphQL interfaces as API

lib/internal/Magento/Framework/GraphQl/Config/ConfigElementFactoryInterface.php
lib/internal/Magento/Framework/GraphQl/Config/ConfigElementInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/FieldInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/OutputFieldInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/TypeInterface.php
lib/internal/Magento/Framework/GraphQl/Config/Element/UnionInterface.php
lib/internal/Magento/Framework/GraphQl/ConfigInterface.php
lib/internal/Magento/Framework/GraphQl/Query/ErrorHandlerInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/FieldEntityAttributesInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Argument/SearchCriteria/ArgumentApplierInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchRequestItemInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchServiceContractResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/ContextInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/IdentityInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/ResolveRequestInterface.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/TypeResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Query/ResolverInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/SchemaGeneratorInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Entity/MapperInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Enum/DataMapperInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/InputTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/Output/ElementMapper/FormatterInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/OutputTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/Type/WrappedTypeInterface.php
lib/internal/Magento/Framework/GraphQl/Schema/TypeInterface.php
lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php


Reference classes/interfaces also marked api
lib/internal/Magento/Magento/Framework/GraphQl/Schema/Type/ResolveInfo.php
lib/internal/Magento/Framework/GraphQl/Config/Element/Field.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchResponse.php
lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php
lib/internal/Magento/Framework/GraphQl/Schema.php

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#32028

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->
CC @sivaschenko 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

eduard13 commented:
@magento run all tests

mrtuvn commented:
@magento run Database Compare, Functional Tests B2B, Static Tests

mrtuvn commented:
@magento run Static Tests, Functional Tests B2B, Database Compare

eduard13 commented:
@magento run Static Tests, Functional Tests B2B, Database Compare

magento-engcom-team commented:
Hi @eduard13, thank you for the review.
[ENGCOM-8782](https://jira.corp.magento.com/browse/ENGCOM-8782) has been created to process this Pull Request

eduard13 commented:
@magento run Database Compare

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@sivaschenko interfaces mark (non-api) in my description seem got issue when add api annotation
related with problem Magento\Test\Integrity\PublicCodeTest->testAllPHPClassesReferencedFromPublicClassesArePublic()

mrtuvn commented:
update + fix static
@magento run all tests

mrtuvn commented:
@magento run WebAPI Tests

mrtuvn commented:
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

engcom-Oscar commented:
Hi @mrtuvn !
Looks like you missed lib/internal/Magento/Framework/GraphQl/Query/Resolver/BatchServiceContractResolverInterface.php.
Please recheck.

mrtuvn commented:
Nice catch! I will update

mrtuvn commented:
cc: @engcom-Oscar corrected my mistakes

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Unit Tests, Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests EE, Functional Tests B2B

mrtuvn commented:
@gabrieldagama should this PR target to branch api-changes-develop

gabrieldagama commented on lib/internal/Magento/Framework/GraphQlSchemaStitching/GraphQlReader/TypeMetaReaderInterface.php:
```diff
@@ -18,5 +20,5 @@ interface TypeMetaReaderInterface
      * @return array|null
      */
-    public function read(\GraphQL\Type\Definition\Type $typeMeta)  : array;
+    public function read(\GraphQL\Type\Definition\Type $typeMeta): ?array;
```
Please, do not change the return type of this function. This is backward incompatible and may create other problems.

mrtuvn committed:
Mark Framework GraphQL interfaces as API


update


update


update


update


update


Update TypeMetaReaderInterface.php

Revert change the return type of this function

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run WebAPI Tests, Unit Tests, Static Tests, Sample Data Tests EE, Sample Data Tests B2B, Functional Tests EE, Functional Tests CE, Functional Tests B2B

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
cc @@gabrieldagama  updated PR branch

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Unit Tests, Static Tests, Integration Tests, Functional Tests CE, Functional Tests EE, Functional Tests B2B

gabrieldagama commented:
@mrtuvn, please ignore the failures, for now, they are related to the fact that `api-changes-develop` is not up to date with `2.4-develop`, it will be fixed soon.

gabrieldagama commented:
@magento import code to https://github.com/magento-engcom/magento2ce

magento-engcom-team commented:
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32101`.
